src/engine/loopingcontrol.cpp

```diff
@@ -513,20 +577,41 @@ void LoopingControl::slotLoopOut(double val) {
         setLoopingEnabled(true);
     }
     //qDebug() << "set loop_out to " << loopSamples.end;
+
+    m_bLoopSetSinceTrackLoaded = true;
+}
+
+void LoopingControl::slotLoopOutGoto(double pressed) {
+    if (pressed > 0.0) {
+        seekAbs(static_cast<double>(
+            m_loopSamples.getValue().end));
+    }
 }
 
 void LoopingControl::slotLoopExit(double val) {
     if (!m_pTrack || val <= 0.0) {
         return;
     }
- 
+
     // If we're looping, stop looping
     if (m_bLoopingEnabled) {
         setLoopingEnabled(false);
     }
 }
 
-void LoopingControl::slotReloopExit(double val) {
+void LoopingControl::slotReloop(double pressed) {
+    if (!m_pTrack || pressed <= 0.0) {
+        return;
+    }
+
+    if (m_bLoopingEnabled) {
+        slotReloopCue(1);
+    } else {
+        slotReloopToggle(1);
+    }
```

Comment:
As far as I know this is a novel behavior that no other software or hardware has. The use case is to set a loop in headphones without worrying about how the phrasing lines up with the track(s) currently audible in the master mix, press this button, then just press play at the right moment to start mixing in the loop.